flake.nix

```diff
@@ -29,24 +17,36 @@
     flake-parts.lib.mkFlake { inherit inputs; } {
       imports = [
         ./flake-module.nix
+        flake-parts.flakeModules.partitions
       ];
 
       systems = import
         inputs.systems;
 
+      partitions.dev = {
+        extraInputsFlake = ./dev;
+        module.imports = [ ./dev/flake-module.nix ];
+      };
+
+      partitionedAttrs = {
+        apps = "dev";
+        devShells = "dev";
+        formatter = "dev";
+        templates = "dev";
+      };
+
       perSystem =
-        { config
-        , self'
+        { pkgs
         , system
         , ...
-        }: let
-          pkgs = import inputs.nixpkgs {
+        }:
+        {
+          _module.args.pkgs = import inputs.nixpkgs {
```

Comment:
This change let's us skip the impure devShell used by direnv, since the pkgs attribute is shared among all the configuration (including the dev shell).